client/src/pages/spools/create.tsx

```diff
@@ -49,10 +50,17 @@ export const SpoolCreate: React.FC<IResourceComponentsProps & CreateOrCloneProps
     return {
       label: label,
       value: item.id,
+	    weight: item.weight,
+	    spool_weight: item.spool_weight
     };
   });
   filamentOptions?.sort((a, b) => a.label.localeCompare(b.label, undefined, { sensitivity: "base" }));
 
+  const [used_weight, set_used_weight] = useSavedState("create-used_weight", 0);
+
+  const selected_filament_id = Form.useWatch('filament_id', form);
+  const selected_filament = filamentOptions?.find(obj => {return obj.value === selected_filament_id});
```

Comment:
The naming convention in the typescript is lowerCamelCase, snake_case is only for the python part.